superset/bin/superset

```diff
@@ -4,7 +4,82 @@ from __future__ import division
 from __future__ import print_function
 from __future__ import unicode_literals
 
-from superset.cli import manager
+import argparse
+import sys
+
+import gunicorn.app.base
+
+
+class GunicornSupersetApplication(gunicorn.app.base.BaseApplication):
+    def __init__(self, address, port, workers, timeout):
+        self.options = {
+            'bind': '%s:%s' % (address, port),
+            'workers': workers,
+            'timeout': timeout,
+            'limit-request-line': 0,
+            'limit-request-field_size': 0
+        }
+
+        super(GunicornSupersetApplication, self).__init__()
+
+    def load_config(self):
+        config = dict([(key, value) for key, value in self.options.iteritems()
+                       if key in self.cfg.settings and value is not None])
+        for key, value in config.iteritems():
+            self.cfg.set(key.lower(), value)
+
+    def load(self):
+        from superset import app
+
+        return app
+
+
+def run_server():
+    parser = argparse.ArgumentParser(description='Run gunicorn for superset')
+    subparsers = parser.add_subparsers()
+    gunicorn_parser = subparsers.add_parser('runserver')
+
+    gunicorn_parser.add_argument(
+        '-d', '--debug', action='store_true',
+        help='Start the web server in debug mode')
+    gunicorn_parser.add_argument(
+        '-a', '--address', type=str, default='127.0.0.1',
+        help='Specify the address to which to bind the web server')
+    gunicorn_parser.add_argument(
+        '-p', '--port', type=int, default=8088,
```

Comment:
> I hope it's not :) Care to open a PR?

The solution is not easy it seems. The idea here appears to be to avoid importing `superset` before the Gunicorn application is loaded. So at the point where the flag defaults are set we cannot access the Superset config file yet.

I think it may be fine to retire these settings. Gunicorn can be configured via a config file. Superset could have a separate Gunicorn config file instead that would allow setting these web-server-specific parameters in one logical place. Plus it would give the user a chance to set many other Gunicorn settings that cannot be controlled now, such as the [SSL certificate](http://docs.gunicorn.org/en/stable/settings.html#ssl).

What do you think?